### PR TITLE
Create peewee models

### PR DESCRIPTION
**Pull request by mc18g13**

### CODE REVIEW

File: boxwise_flask/graph_ql/type_defs.py [hunk 1]
```diff
@@ -1,11 +1,107 @@
...
 type_defs = gql(
     """
-    type Base {
-        id: Int
+    type QRCode {
+        id: Int!
+        code: String!
+        createdOn: Int
+        createdBy: String!
+        lastModifiedOn: Int
```
Comment (HaGuesto) [inline boxwise_flask/graph_ql/type_defs.py:18]: A QR-code cannot be modified. Not sure where this comes from?

File: boxwise_flask/graph_ql/type_defs.py [hunk 2]
```diff
@@ -1,11 +1,107 @@
...
+        box: Box
+    }
+
+    type Product {
+        id: Int!
+        name: String!
+        category: ProductCategory!
```
Comment (HaGuesto) [inline boxwise_flask/graph_ql/type_defs.py:25]: please keep the naming consistent or do you have a reason?

File: boxwise_flask/graph_ql/type_defs.py [hunk 3]
```diff
@@ -1,11 +1,107 @@
...
+        sizeRange: SizeRange!
+        sizes: [String!]!
+        base: Base!
+        price: Float
+        gender: ProductGender
```
Comment (HaGuesto) [inline boxwise_flask/graph_ql/type_defs.py:30]: please keep the naming consistent or do you have a reason?

File: boxwise_flask/graph_ql/type_defs.py [hunk 4]
```diff
@@ -1,11 +1,107 @@
...
+        lastModifiedOn: Int
+        box: Box
+    }
+
+    type Product {
+        id: Int!
+        name: String!
+        category: ProductCategory!
+        sizeRange: SizeRange!
+        sizes: [String!]!
```
Comment (HaGuesto) [inline boxwise_flask/graph_ql/type_defs.py:27]: what is this field doing? Should this not be covered by sizeRange?

File: boxwise_flask/models/base.py [hunk 5]
```diff
@@ -1,12 +1,76 @@
...
-    organisation_id = IntegerField()
-    name = CharField()
-    currency_name = CharField(column_name="currencyname")
+    name = CharField(null=True)
+    currency_name = CharField(
+        column_name="currencyname", constraints=[SQL("DEFAULT 'Tokens'")]
+    )
+
+    adult_age = IntegerField(constraints=[SQL("DEFAULT 15")])
+    bicycle = IntegerField(constraints=[SQL("DEFAULT 0")])
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:24]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 6]
```diff
@@ -1,12 +1,76 @@
...
+    bicycle_closingtime = CharField(constraints=[SQL("DEFAULT '17:30'")], null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:25]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 7]
```diff
@@ -1,12 +1,76 @@
...
+    bicycle_closingtime_saturday = CharField(
+        constraints=[SQL("DEFAULT '16:30'")], null=True
+    )
+    bicyclerenttime = IntegerField(constraints=[SQL("DEFAULT 120")])
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:29]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 8]
```diff
@@ -1,12 +1,76 @@
...
+    )
+    delete_inactive_users = IntegerField(constraints=[SQL("DEFAULT 30")])
+    deleted = DateTimeField(null=True)
+    dropcapadult = IntegerField(constraints=[SQL("DEFAULT 99999")])
+    dropcapchild = IntegerField(constraints=[SQL("DEFAULT 99999")])
+    dropsperadult = CharField(constraints=[SQL("DEFAULT '100'")])
+    dropsperchild = CharField(constraints=[SQL("DEFAULT '100'")])
+    extraportion = IntegerField(constraints=[SQL("DEFAULT 0")], null=True)
+    familyidentifier = CharField(constraints=[SQL("DEFAULT 'Container'")])
+    food = IntegerField(constraints=[SQL("DEFAULT 0")])
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:47]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 9]
```diff
@@ -1,12 +1,76 @@
...
+    idcard = IntegerField(constraints=[SQL("DEFAULT 0")])
+    laundry = IntegerField(constraints=[SQL("DEFAULT 0")])
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:49]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 10]
```diff
@@ -1,12 +1,76 @@
...
+    laundry_cyclestart = DateField(default=datetime.now(), null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:50]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 11]
```diff
@@ -1,12 +1,76 @@
...
+    market = IntegerField(constraints=[SQL("DEFAULT 1")])
+    maxfooddrops_adult = IntegerField(constraints=[SQL("DEFAULT 25")], null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:52]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 12]
```diff
@@ -1,12 +1,76 @@
...
+    maxfooddrops_child = IntegerField(constraints=[SQL("DEFAULT 25")], null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:53]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 13]
```diff
@@ -1,12 +1,76 @@
...
+    )
+    resettokens = IntegerField(constraints=[SQL("DEFAULT 0")], null=True)
+    schedulebreak = CharField(constraints=[SQL("DEFAULT '1'")])
+    schedulebreakduration = CharField(constraints=[SQL("DEFAULT '1'")])
+    schedulebreakstart = CharField(constraints=[SQL("DEFAULT '13:00'")])
+    schedulestart = CharField(constraints=[SQL("DEFAULT '11:00'")])
+    schedulestop = CharField(constraints=[SQL("DEFAULT '17:00'")])
+    scheduletimeslot = CharField(constraints=[SQL("DEFAULT '0.5'")])
+    seq = IntegerField()
+    workshop = IntegerField(constraints=[SQL("DEFAULT 0")])
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:73]: not needed anymore.

File: boxwise_flask/models/base.py [hunk 14]
```diff
@@ -31,6 +95,6 @@ def get_for_organisation(org_id):
         return Base.select().where(Base.organisation_id == org_id)
 
     @staticmethod
-    def get_from_id(camp_id):
-        camp = Base.select().where(Base.id == camp_id).get()
-        return camp
+    def get_from_id(base_id):
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:98]: I want a test for this. I will write down all the model tests in [here](https://docs.google.com/spreadsheets/d/1sDhSsaVwNxAhGn1VYACDPVepZpvlwTkgy9nmeTaRy2Q/edit?usp=sharing)

File: boxwise_flask/models/base.py [hunk 15]
```diff
@@ -1,12 +1,76 @@
...
+    familyidentifier = CharField(constraints=[SQL("DEFAULT 'Container'")])
+    food = IntegerField(constraints=[SQL("DEFAULT 0")])
+    idcard = IntegerField(constraints=[SQL("DEFAULT 0")])
+    laundry = IntegerField(constraints=[SQL("DEFAULT 0")])
+    laundry_cyclestart = DateField(default=datetime.now(), null=True)
+    market = IntegerField(constraints=[SQL("DEFAULT 1")])
+    maxfooddrops_adult = IntegerField(constraints=[SQL("DEFAULT 25")], null=True)
+    maxfooddrops_child = IntegerField(constraints=[SQL("DEFAULT 25")], null=True)
+    modified = DateTimeField(null=True)
+    modified_by = ForeignKeyField(
```
Comment (HaGuesto) [inline boxwise_flask/models/base.py:55]: Why does this look so ugly?

File: boxwise_flask/models/box.py [hunk 16]
```diff
@@ -1,27 +1,77 @@
...
+from boxwise_flask.models.location import Location
+from boxwise_flask.models.product import Product
+from boxwise_flask.models.size import Size
+from boxwise_flask.models.user import User
+
+from .qr_code import QRCode
 
 
 class Box(db.Model):
+    _gender = CharField(null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:24]: not needed anymore.

File: boxwise_flask/models/box.py [hunk 17]
```diff
@@ -1,27 +1,77 @@
...
+    _size = CharField(null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:25]: not needed anymore.

File: boxwise_flask/models/box.py [hunk 18]
```diff
@@ -1,27 +1,77 @@
...
+    _type = CharField(null=True)
+    box_id = CharField(constraints=[SQL("DEFAULT ''")], index=True)
+    box_state = ForeignKeyField(
+        column_name="box_state_id",
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:29]: not needed anymore.

File: boxwise_flask/models/box.py [hunk 19]
```diff
@@ -1,27 +1,77 @@
...
+from boxwise_flask.models.size import Size
+from boxwise_flask.models.user import User
+
+from .qr_code import QRCode
 
 
 class Box(db.Model):
+    _gender = CharField(null=True)
+    _size = CharField(null=True)
+    _type = CharField(null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:26]: not needed anymore.

File: boxwise_flask/models/box.py [hunk 20]
```diff
@@ -1,27 +1,77 @@
...
+        constraints=[SQL("DEFAULT 1")],
+        field="id",
+        model=BoxState,
+    )
+    comments = TextField()
+    created = DateTimeField(null=True)
+    created_by = ForeignKeyField(
+        column_name="created_by", field="id", model=User, null=True
+    )
+    deleted = DateTimeField(default=datetime.now())
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:39]: deleted fields should always include `null=true` and `default=null`.

File: boxwise_flask/models/box.py [hunk 21]
```diff
@@ -1,27 +1,77 @@
...
+    items = IntegerField()
+    location = ForeignKeyField(column_name="location_id", field="id", model=Location)
+    modified = DateTimeField(null=True)
+    modified_by = ForeignKeyField(
+        backref="cms_users_modified_by_set",
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:44]: Why is this so ugly?

File: boxwise_flask/models/box.py [hunk 22]
```diff
@@ -1,27 +1,77 @@
...
+        column_name="modified_by",
+        field="id",
+        model=User,
+        null=True,
+    )
+    ordered = DateTimeField(null=True)
+    ordered_by = ForeignKeyField(
+        backref="cms_users_ordered_by_set",
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:52]: why is this so ugly?

File: boxwise_flask/models/box.py [hunk 23]
```diff
@@ -1,27 +1,77 @@
...
+        column_name="ordered_by",
+        field="id",
+        model=User,
+        null=True,
+    )
+    picked = IntegerField(null=True)
+    picked_by = ForeignKeyField(
+        backref="cms_users_picked_by_set",
```
Comment (HaGuesto) [inline boxwise_flask/models/box.py:60]: why is this so ugly?

File: boxwise_flask/models/language.py [hunk 24]
```diff
@@ -0,0 +1,17 @@
+from peewee import SQL, CharField, IntegerField
```
Comment (HaGuesto) [inline boxwise_flask/models/language.py:1]: I do not think we need this whole file

File: boxwise_flask/models/location.py [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+    )
+    deleted = DateTimeField(null=True)
+    is_donated = IntegerField(constraints=[SQL("DEFAULT 0")])
+    is_lost = IntegerField(constraints=[SQL("DEFAULT 0")])
+    is_market = IntegerField(constraints=[SQL("DEFAULT 0")])
+    is_scrap = IntegerField(constraints=[SQL("DEFAULT 0")])
+    label = CharField()
+    modified = DateTimeField(null=True)
+    modified_by = ForeignKeyField(
+        backref="cms_users_modified_by_set",
```
Comment (HaGuesto) [inline boxwise_flask/models/location.py:31]: why is this so ugly?

File: boxwise_flask/models/product.py [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+from boxwise_flask.models.base import Base
+from boxwise_flask.models.product_category import ProductCategory
+from boxwise_flask.models.product_gender import ProductGender
+from boxwise_flask.models.size_range import SizeRange
+from boxwise_flask.models.user import User
+
+
+class Product(db.Model):
+    amountneeded = FloatField(constraints=[SQL("DEFAULT 1")])
+    camp = ForeignKeyField(column_name="camp_id", field="id", model=Base, null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/product.py:20]: This should be named base

File: boxwise_flask/models/product.py [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+    category = ForeignKeyField(
+        column_name="category_id", field="id", model=ProductCategory, null=True
+    )
+    comments = CharField(null=True)
+    created = DateTimeField(null=True)
+    created_by = ForeignKeyField(
+        column_name="created_by", field="id", model=User, null=True
+    )
+    deleted = DateTimeField(null=True)
+    gender = ForeignKeyField(
```
Comment (HaGuesto) [inline boxwise_flask/models/product.py:30]: can we name this productGender to stay consistent

File: boxwise_flask/models/product.py [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+        column_name="gender_id", field="id", model=ProductGender, null=True
+    )
+    maxperadult = IntegerField(null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/product.py:33]: not needed anymore

File: boxwise_flask/models/product.py [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+    maxperchild = IntegerField(null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/product.py:34]: not needed anymore

File: boxwise_flask/models/product.py [hunk 30]
```diff
@@ -0,0 +1,51 @@
...
+    modified = DateTimeField(null=True)
+    modified_by = ForeignKeyField(
+        backref="cms_users_modified_by_set",
```
Comment (HaGuesto) [inline boxwise_flask/models/product.py:37]: why? :)

File: boxwise_flask/models/product.py [hunk 31]
```diff
@@ -0,0 +1,51 @@
...
+        column_name="modified_by",
+        field="id",
+        model=User,
+        null=True,
+    )
+    name = CharField()
+    sizegroup = ForeignKeyField(
```
Comment (HaGuesto) [inline boxwise_flask/models/product.py:44]: Please name sizeRange

File: boxwise_flask/models/size.py [hunk 32]
```diff
@@ -0,0 +1,40 @@
...
+    label = CharField()
+    modified = DateTimeField(null=True)
+    modified_by = ForeignKeyField(
+        backref="cms_users_modified_by_set",
+        column_name="modified_by",
+        field="id",
+        model=User,
+        null=True,
+    )
+    portion = IntegerField(null=True)
```
Comment (HaGuesto) [inline boxwise_flask/models/size.py:22]: not needed anymore

File: boxwise_flask/models/size.py [hunk 33]
```diff
@@ -0,0 +1,40 @@
...
+    seq = IntegerField(null=True)
+    sizegroup = ForeignKeyField(
```
Comment (HaGuesto) [inline boxwise_flask/models/size.py:24]: sizeRange

File: boxwise_flask/models/user.py [hunk 34]
```diff
@@ -1,36 +1,67 @@
...
+    modified = DateTimeField(null=True)
+    modified_by = ForeignKeyField(
+        backref="cms_users_modified_by_set",
+        column_name="modified_by",
+        field="id",
+        model="self",
+        null=True,
+    )
+    name = CharField(constraints=[SQL("DEFAULT ''")])
+    pass_ = CharField(column_name="pass", constraints=[SQL("DEFAULT ''")])
```
Comment (HaGuesto) [inline boxwise_flask/models/user.py:42]: why is here an underscore?

File: boxwise_flask/models/usergroup.py [hunk 35]
```diff
@@ -0,0 +1,36 @@
...
+    IntegerField,
+)
+
+from boxwise_flask.db import db
+from boxwise_flask.models.organisation import Organisation
+from boxwise_flask.models.usergroup_access_level import UsergroupAccessLevel
+
+
+class Usergroup(db.Model):
+    allow_borrow_adddelete = IntegerField()
```
Comment (HaGuesto) [inline boxwise_flask/models/usergroup.py:16]: not needed anymore

File: boxwise_flask/models/usergroup.py [hunk 36]
```diff
@@ -0,0 +1,36 @@
...
+    allow_laundry_block = IntegerField(constraints=[SQL("DEFAULT 0")])
```
Comment (HaGuesto) [inline boxwise_flask/models/usergroup.py:17]: why is this so ugly?

File: boxwise_flask/models/usergroup.py [hunk 37]
```diff
@@ -0,0 +1,36 @@
...
+    allow_laundry_startcycle = IntegerField(constraints=[SQL("DEFAULT 0")])
```
Comment (HaGuesto) [inline boxwise_flask/models/usergroup.py:18]: not needed anymore

File: boxwise_flask/models/usergroup_module_access.py [hunk 38]
```diff
@@ -0,0 +1,19 @@
...
+from boxwise_flask.db import db
+from boxwise_flask.models.base_module import BaseModule
+from boxwise_flask.models.usergroup import Usergroup
+
+
+class UsergroupModuleAccess(db.Model):
+    cms_functions = ForeignKeyField(
+        column_name="cms_functions_id", field="id", model=BaseModule, null=True
+    )
+    cms_usergroups = ForeignKeyField(
```
Comment (HaGuesto) [inline boxwise_flask/models/usergroup_module_access.py:12]: other naming please

File: boxwise_flask/models/usergroup_module_access.py [hunk 39]
```diff
@@ -0,0 +1,19 @@
+from peewee import ForeignKeyField
+
+from boxwise_flask.db import db
+from boxwise_flask.models.base_module import BaseModule
+from boxwise_flask.models.usergroup import Usergroup
+
+
+class UsergroupModuleAccess(db.Model):
+    cms_functions = ForeignKeyField(
```
Comment (HaGuesto) [inline boxwise_flask/models/usergroup_module_access.py:9]: other naming please

File: boxwise_flask/routes.py [hunk 40]
```diff
@@ -79,7 +79,7 @@ def graphql_server():
     # Note: Passing the request to the context is optional.
     # In Flask, the current request is always accessible as flask.request
 
-    debug_graphql = bool(os.getenv("DEBUG_GRAPHQL", False))
+    debug_graphql = True  # bool(os.getenv("DEBUG_GRAPHQL", False))
```
Comment (HaGuesto) [inline boxwise_flask/routes.py:82]: why?

File: boxwise_flask/models/box.py (hunk 20 above)
Comment (mc18g13) [reply boxwise_flask/models/box.py:39]: I had this as a question. I was confused why it wasn't already like this.

File: boxwise_flask/models/language.py (hunk 24 above)
Comment (mc18g13) [reply boxwise_flask/models/language.py:1]: Can I remove this from the user table also?

File: boxwise_flask/models/user.py (hunk 34 above)
Comment (mc18g13) [reply boxwise_flask/models/user.py:42]: pass is a reserved keyword in python

File: boxwise_flask/models/usergroup.py [hunk 41]
```diff
@@ -0,0 +1,36 @@
...
+    created = DateTimeField(null=True)
+    created_by = DeferredForeignKey("User")
+    deleted = DateTimeField(null=True)
+    label = CharField(null=True)
+    modified = DateTimeField(null=True)
+    modified_by = DeferredForeignKey("User")
```
[the remark this answers is not in the document]
Comment (mc18g13) [reply boxwise_flask/models/usergroup.py:24]: it means it'll automatically pick up the reference during the import of other tables. It helps solve circular reference. say if a table refers to itself.

File: boxwise_flask/routes.py (hunk 40 above)
Comment (mc18g13) [reply boxwise_flask/routes.py:82]: My bad this is still a draft. Graphql gives little information when models don't line up correctly. just wanted to make sure I was getting all the information

**Comment by aerinsol** [comment]
Is it possible move this to the new repo?

**Comment by HaGuesto** [comment]
closed due to https://github.com/boxwise/boxtribute/pull/1

**Pull request closed without merging**